File: examples/android/ProfileViewModel.kt
// Paired with: agents/android/code-reviewer/agent.md
// This file contains intentional issues for the code reviewer agent to find.
// Do not use this as production code.

package com.example.profile

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.launch

data class User(val id: String, val name: String, val bio: String)

class UserRepository {
    suspend fun getUser(id: String): User = User(id, "Alice", "Developer")
    suspend fun updateUser(user: User) {}
}

class ProfileViewModel : ViewModel() {

    val user = MutableLiveData<User>()

    fun loadUser(userId: String) {
        GlobalScope.launch {
            val result = UserRepository().getUser(userId)
            user.value = result
        }
    }

    fun saveProfile(name: String, bio: String) {
        if (name == null || name.isEmpty()) return
        val updated = User(id = user.value!!.id, name = name, bio = bio)
        GlobalScope.launch {
            UserRepository().updateUser(updated)
        }
    }
}

File: examples/ios/ProfileViewModel.swift
// Paired with: agents/ios/swift-reviewer/agent.md
// This file contains intentional issues for the Swift reviewer agent to find.
// Do not use this as production code.

import Foundation
import UIKit
import Combine

struct User {
    var id: String
    var name: String
    var bio: String
    var avatarUpdated: Bool = false
}

protocol ProfileServiceProtocol {
    func fetchUser(id: String, completion: @escaping (Result<User, Error>) -> Void)
    func upload(data: Data, completion: @escaping (Bool) -> Void)
}

class ProfileService: ProfileServiceProtocol {
    func fetchUser(id: String, completion: @escaping (Result<User, Error>) -> Void) {
        DispatchQueue.global().asyncAfter(deadline: .now() + 1) {
            completion(.success(User(id: id, name: "Alice", bio: "Developer")))
        }
    }

    func upload(data: Data, completion: @escaping (Bool) -> Void) {
        DispatchQueue.global().asyncAfter(deadline: .now() + 2) {
            completion(true)
        }
    }
}

class ProfileViewModel: ObservableObject {
    @Published var user: User?
    @Published var error: String?

    let service = ProfileService()

    func loadUser(id: String) {
        service.fetchUser(id: id) { result in
            switch result {
            case .success(let user):
                self.user = user
            case .failure(let error):
                self.error = error.localizedDescription
            }
        }
    }

    func uploadAvatar(image: UIImage) {
        let data = image.jpegData(compressionQuality: 1.0)!
        service.upload(data: data) { [unowned self] success in
            if success { self.user?.avatarUpdated = true }
        }
    }
}
